spring-pulsar/src/main/java/org/springframework/pulsar/listener/DefaultPulsarMessageListenerContainer.java

```diff
@@ -439,6 +454,21 @@ public void run() {
 						handleAcks(messages);
 					}
 				}
+				if (isPaused()) {
+					try {
+						// try acquiring the lock.
+						DefaultPulsarMessageListenerContainer.this.lockOnPause.lock();
```

Comment:
Did you intend to have this w/in the try block?

Reply:
good catch.